### ISSUE TITLE
Parity error in infinite pending block

### ISSUE THREAD

**Issue by dareniott** [paritytech/parity-ethereum#9349]
> - **Which Parity version?**: 1.11.8-stable
> - **Which operating system?**: Linux
> - **How installed?**: docker
> - **Are you fully synchronized?**:  yes
> - **Which network are you connected to?**: ethereum
> - **Did you try to restart the node?**: yes

Runs fine with command line: 
`--auto-update none --no-download --relay-set strict --force-sealing --cache-size-db 1024 --pruning=archive --tracing=on --min-peers=100 --max-peers=200 --jsonrpc-interface all --jsonrpc-hosts all --no-ui --ws-interface=all --ws-apis=all --ws-hosts=all --tx-queue-gas off --reseal-on-txs all --tx-queue-size 1000000 --reseal-min-period=1
`

Gives the following error with the addition of the --infinite-pending-block option:
```
Thread 'IO Worker #1' panicked at 'Integer overflow when casting U256', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ethereum-types-0.3.2/src/uint.rs:32 
Aug 13 13:43:42 parity-1 default[parity-566689cd74-l58gj]: This is a bug. Please report it at: 
Aug 13 13:43:42 parity-1 default[parity-566689cd74-l58gj]:     https://github.com/paritytech/parity/issues/new 
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC Starting Parity/v1.11.8-stable-92776e4-20180728/x86_64-linux-gnu/rustc1.27.2
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC Keys path /root/.local/share/io.parity.ethereum/keys/Foundation
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC DB path /root/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC Path to dapps /root/.local/share/io.parity.ethereum/dapps
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC State DB configuration: archive +Trace
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC Operating mode: active
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC Warning: Warp Sync is disabled because tracing is turned on. 
Aug 13 13:44:08 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:08 UTC Configured for Foundation using Ethash engine
Aug 13 13:44:09 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:09 UTC Removed existing file '/root/.local/share/io.parity.ethereum/jsonrpc.ipc'. 
Aug 13 13:44:09 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:09 UTC Updated conversion rate to Ξ1 = US$286.03 (16648269 wei/gas)
Aug 13 13:44:14 parity-1 default[parity-566689cd74-l58gj]: 2018-08-13 17:44:14 UTC Public node URL: enode://45accd686bbe2a874d61ba34b58375a34ea671d9e3c7148528fb6c7dd8f9a1b9c09b0dfa0c3f328b5dd3e8f15edf4b350577150f34e6b52f60d3dcfff72eb3a6@10.42.0.78:30303
Aug 13 13:44:15 parity-1 default[parity-566689cd74-l58gj]: ==================== 
Aug 13 13:44:15 parity-1 default[parity-566689cd74-l58gj]: stack backtrace: 
Aug 13 13:44:15 parity-1 default[parity-566689cd74-l58gj]:    0:     0x555a095a080c - <no info> 
Aug 13 13:44:15 parity-1 default[parity-566689cd74-l58gj]:    1:     0x555a0959fe32 - <no info> 
Aug 13 13:44:15 parity-1 default[parity-566689cd74-l58gj]:    2:     0x555a08cdda0a - <no info> 
```

**Comment by kavu** [comment]
I tried to reproduce it.

On `Parity/v1.11.8-stable-c754a02-20180725/x86_64-macos/rustc1.27.2` minimal command to crash Parity is `parity --force-sealing  --tracing=on --infinite-pending-block` with no previous "data" (chains, keys, cache, network, etc.) present. 

And I can't repeat it on current `Parity-Ethereum/v2.1.0-unstable-c21c19bd6-20180814/x86_64-macos/rustc1.28.0`. Will investigate it further.

**Comment by HCastano** [comment]
I believe this was fixed in #9221, which is why it is reproducible in the v1.11.8-stable release but not the v2.1.0 build you tried.

**Comment by Tbaut** [comment]
This fix will be backported, you should have it in the upcoming 1.11.9
